pkg/wshrpc/wshremote/wshremote.go

```
@@ -186,37 +181,337 @@ func (impl *ServerImpl) remoteStreamFileInternal(ctx context.Context, data wshrp
+	cleanedPath := filepath.Clean(wavebase.ExpandHomeDirSafe(path))
+	finfo, err := os.Stat(cleanedPath)
+	if err != nil {
+		return wshutil.SendErrCh[[]byte](fmt.Errorf("cannot stat file %q: %w", path, err))
+	}
+	pipeReader, pipeWriter := io.Pipe()
+	tarWriter := tar.NewWriter(pipeWriter)
+	timeout := time.Millisecond * 100
+	if opts.Timeout > 0 {
+		timeout = opts.Timeout
+	}
+	readerCtx, _ := context.WithTimeout(context.Background(), timeout)
```

golangci-lint (1.62.2)

lostcancel: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak

(govet)